sympy/solvers/solveset.py

```diff
@@ -975,6 +982,438 @@ def _solveset(f, symbol, domain, _check=False):
                       if isinstance(s, RootOf)
                       or domain_check(fx, symbol, s)])
 
+            if domain.is_subset(S.Reals) and _is_logarithmic(f, symbol):
```

Comment:
This check is done so as to remove unwanted solutions form the original result.
This method is implemented rather than using `checksol` because there are some unwanted solution that would still creep into the actual result.

Take for example `log(3*x) - log(-x + 1) - log(4*x + 1)`, the result in general is `-1/2 and 1/2` but the negative solution is only valid if we consier complex valued logairthm but since we are concerned with the Real domain as of now only natural logairthm will work which is undefined for negative as well as for `0` therefore `-1/2` can't be the solution in real doamin.
If the result is verified using `checksol` the subsitution would cause the cancelling of `I*pi` and hence it would return `True`, making `-1/2` as the solution

In this method each solution is tried in each term of the expression, if its result is not in a real form then it is not the solution (This is specifically beneficial for checking for logarithmic expressions).